Vector center
Vector normal
Vector upperLeft
Vector lowerLeft
Vector upperRight
Vector lowerRight
